ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -12,63 +12,69 @@ export default class PermissionPageContainerContent extends PureComponent {
     selectedPermissions: PropTypes.object.isRequired,
     permissionsDescriptions: PropTypes.object.isRequired,
     onPermissionToggle: PropTypes.func.isRequired,
-    selectedAccount: PropTypes.object,
+    selectedIdentities: PropTypes.array,
     redirect: PropTypes.bool,
     permissionRejected: PropTypes.bool,
   }
 
   static defaultProps = {
     redirect: null,
     permissionRejected: null,
-    selectedAccount: {},
+    selectedIdentities: [],
   }
 
   static contextTypes = {
     t: PropTypes.func,
   }
 
-  renderAccountInfo = (account) => {
+  renderRedirectVisual () {
+    const { permissionRejected, selectedIdentities, domainMetadata } = this.props
     return (
-      <div className="permission-approval-visual__account-info">
-        <div className="permission-approval-visual__account-info__label">
-          { account.label }
+      <div className="permission-approval-visual__redirect">
+        <IconWithFallBack icon={domainMetadata.icon} name={domainMetadata.name} />
+        <div className="permission-approval-visual__redirect-center">
+          { permissionRejected
+            ? <span className="permission-approval-visual__reject" ><i className="fa fa-times-circle" /></span>
+            : <span className="permission-approval-visual__check" />
+          }
+          <img className="permission-approval-visual__broken-line" src="/images/broken-line.svg" />
         </div>
-        <div className="permission-approval-visual__account-info__address">
-          { account.truncatedAddress }
+        <div className="permission-approval-visual__identicon-container">
+          <div className="permission-approval-visual__identicon-border">
+            <Identicon
+              className="permission-approval-visual__identicon"
+              address={selectedIdentities[0].address}
+              diameter={54}
+            />
+          </div>
         </div>
       </div>
     )
   }
 
+  renderNonRedirectVisual () {
+    const { requestMetadata, domainMetadata } = this.props
+
+    return (
+      <div>
```

Comment:
This entire component (the dapp icon, name, and origin) should be shared between this page and the previous page from the look of the designs. Maybe it makes sense as a separate component, or pulled up the the above component? (though if it's pulled up, this redirect state/page would also need to be pulled up.

Reply:
This concern was addressed in aafa179f0, with some related changes also included in the later commit 3e9d3e03f